geode-core/src/distributedTest/java/org/apache/geode/management/MemberMXBeanDistributedTest.java

```diff
@@ -82,16 +82,19 @@ public void testBucketCount() {
     server3.invoke(() -> createBuckets(regionName));
     server4.invoke(() -> createBuckets(regionName));
 
-    await().untilAsserted(() -> {
-      final int sumOfBuckets = server1.invoke(() -> getBucketsInitialized()) +
-          server2.invoke(() -> getBucketsInitialized()) +
-          server3.invoke(() -> getBucketsInitialized()) +
-          server4.invoke(() -> getBucketsInitialized());
-      assertEquals("Expected bucket count is 1000, and actual count is " + sumOfBuckets,
-          sumOfBuckets, 1000);
-    });
-
     for (int i = 1; i < 4; i++) {
+
+      final int index = i;
+      await().untilAsserted(() -> {
+        final int sumOfBuckets = server1.invoke(() -> getBucketsInitialized()) +
+            server2.invoke(() -> getBucketsInitialized()) +
+            server3.invoke(() -> getBucketsInitialized()) +
+            server4.invoke(() -> getBucketsInitialized());
+        final int expectedBuckets = index * 1000;
+        assertThat(sumOfBuckets).isEqualTo(expectedBuckets);
+
+      });
```

Comment:
Any reasoning behind moving the assertion block inside the loop?. The actual flakiness was reported for the last assertion, not this one.

Reply:
During testing of reported fault, it was observed incosistency of results, if there were no pause between region creation. 
And also if flakiness still exist, to get more info, at which step fault happens.

Reply:
According to what I see in the last 4 failures within the `CI` for this test, the actual assertion failure happens when we expect 4000 as the total buckets (later in the test), not here:

- https://concourse.apachegeode-ci.info/teams/main/pipelines/apache-develop-main/jobs/DistributedTestOpenJDK11/builds/205
- https://concourse.apachegeode-ci.info/teams/main/pipelines/apache-develop-main/jobs/DistributedTestOpenJDK11/builds/219
- https://concourse.apachegeode-ci.info/teams/main/pipelines/apache-develop-main/jobs/DistributedTestOpenJDK11/builds/234
- https://concourse.apachegeode-ci.info/teams/main/pipelines/apache-develop-main/jobs/DistributedTestOpenJDK11/builds/249

> If there were no pause between region creation.

What do you mean?, if we optimistically hope for the region creation to be completed within the loop we might hit more flakiness. Instead, we should add some logic to double check that all operations have finished on the servers before executing the assertions. The following might help:

```
  private void waitForSilence() {
    DMStats dmStats = cacheRule.getSystem().getDistributionManager().getStats();
    PartitionedRegion region = (PartitionedRegion) cacheRule.getCache().getRegion(REGION_NAME);
    PartitionedRegionStats partitionedRegionStats = region.getPrStats();

    await().untilAsserted(() -> {
      assertThat(dmStats.getReplyWaitsInProgress()).isEqualTo(0);
      assertThat(partitionedRegionStats.getVolunteeringInProgress()).isEqualTo(0);
      assertThat(partitionedRegionStats.getBucketCreatesInProgress()).isEqualTo(0);
      assertThat(partitionedRegionStats.getPrimaryTransfersInProgress()).isEqualTo(0);
      assertThat(partitionedRegionStats.getRebalanceBucketCreatesInProgress()).isEqualTo(0);
      assertThat(partitionedRegionStats.getRebalancePrimaryTransfersInProgress()).isEqualTo(0);
    });
  }
```

Reply:
I mean that we dont execute consecutive creations of regions.
All other stats are not relevant for this test. Reason for this test was to prove previous fault, which was showing value arround 1000, when expected was 4000.
I would not preffer to add additional checks

Reply:
> Reason for this test was to prove previous fault, which was showing value arround 1000, when expected was 4000.

Unless I'm missing something, I don't think this is accurate.
The test itself (before the changes) creates a `PARTITION_PERSISTENT` with `1000` buckets, makes sure all buckets have been created (`PartitionRegionHelper.assignBucketsToPartitions`) on all members and asserts that the total bucket count is `1000`. After that, it creates 3 more `PARTITION_PERSISTENT` with `1000` buckets each, co-located with the original region (`testCreateRegion`), and asserts that the total bucket count at the very end is `4000` (`1000` per region).
The flakiness, I believe, is caused by the fact that the test is not invoking `PartitionRegionHelper.assignBucketsToPartitions` on the 3 regions created within the loop, so when it asserts the results some buckets might have not been created yet.

Reply:
When I say  previous fault, I am referring to the issue, due to which this test was introduced. 
Regarding flakiness, I believe that check with await, after creation of each region will secure that we will not create new region until all processes connected with previous created region are not finished.
But if you not agree I will close this PR.

Reply:
The change to wait within the loop might help, I'm not arguing that, but I still don't think that it will fix the flakiness; it also clutters the test class a lot.
I believe the proper fix would be to force the bucket creation (`PartitionRegionHelper.assignBucketsToPartitions`) within the loop after each region is created, as you're already doing for the initial region (`testCreateRegion`).